core/src/main/java/edu/wpi/grip/core/settings/ProjectSettings.java

```diff
@@ -43,6 +45,30 @@
   private String deployJvmOptions = "-Xmx50m -XX:-OmitStackTraceInFastThrow "
       + "-XX:+HeapDumpOnOutOfMemoryError -XX:MaxNewSize=16m";
 
+  @Setting(label = "Preferred generated language",
+           description = "The preferred language to generate code for")
+  private String preferredGeneratedLanguage = "";
+
+  @Setting(label = "Generated pipeline name",
+           description = "The name of the generated pipeline class")
+  private String generatedPipelineName = "Pipeline";
+
+  @Setting(label = "Generated Python module name",
+           description = "The name of the generated Python module")
+  private String generatedPythonModuleName = "grip";
+
+  @Setting(label = "Generated Java package name",
+           description = "The name of the package the generated Java file is in")
+  private String generatedJavaPackage = "";
+
+  @Setting(label = "Code generation destination directory",
+           description = "The directory where generated code should be saved")
+  private File codegenDestDir = new File(System.getProperty("user.home") + "/GRIP");
```

Comment:
We have the file system mock thing for testing right?

Also, how does this serialize?